airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -597,3 +607,57 @@ def __exit__(self, exctype, excinst, exctb):
             logger = logging.getLogger()
             logger.error(str(excinst), exc_info=True)
         return caught_error
+
+
+def _prune_dict(val: Any, mode='strict'):
+    """
+    Note: this is duplicated from ``airflow.utils.helpers.prune_dict``.  That one is tested
```

Comment:
Could you clarify this message? I'm struggling to understand why it's needed. In which version would you plan to remove it again?

Reply:
It is an internal method so users should be using it so as soon as we make a change to make this provider require Airflow 2.3+ we should be able to remove it without even mentioning in `UPDATING.md` of this provider

Reply:
Yes removing it without notice is perfectly fine to me, but mentioning 2.3, or maybe the PR URL would be nice.